c_src/erl_optics.c

```diff
@@ -36,9 +46,17 @@ static int load(ErlNifEnv *env, void **priv_data, ERL_NIF_TERM load_info) {
     atom_dist = enif_make_atom(env, "dist");
     atom_histo = enif_make_atom(env, "histo");
     atom_quantile = enif_make_atom(env, "quantile");
+    poller = optics_poller_alloc();
```

Comment:
I think you should check for NULL here; optics_poller_alloc() can fail, in which case, I think the Erlang VM should be notified somehow, or loading should fail.